pkg/scheduler/algorithm/priorities/resource_bin_packing.go

```diff
@@ -0,0 +1,240 @@
+
+		}
+
+	case v1.ResourceMemory:
+		{
+			for i := range pod.Spec.Containers {
+				container := &pod.Spec.Containers[i]
+				_, memory := priorityutil.GetNonzeroRequests(&container.Resources.Requests)
+				requested.Memory += memory
+			}
+			for i := range pod.Spec.InitContainers {
+				container := &pod.Spec.InitContainers[i]
+				_, memory := priorityutil.GetNonzeroRequests(&container.Resources.Requests)
+				if requestedInit.Memory < memory {
+					requestedInit.Memory = memory
+				}
+			}
+			if requestedInit.Memory > requested.Memory {
+				requested.Memory = requestedInit.Memory
+			}
+			//println("Memory used: %d, requested: %d, available: %d",
+			//	nodeInfo.NonZeroRequest().Memory, requested.Memory, allocatable.Memory)
+			return float64((nodeInfo.NonZeroRequest().Memory+requested.Memory)*schedulerapi.MaxPriority) / float64(allocatable.Memory)
+		}
+
+	case v1.ResourceEphemeralStorage:
+		{
+			for i := range pod.Spec.Containers {
+				container := &pod.Spec.Containers[i]
+				if quantity, ok := container.Resources.Requests[v1.ResourceEphemeralStorage]; ok {
+					requested.EphemeralStorage += quantity.Value()
+				}
+
+			}
+			for i := range pod.Spec.InitContainers {
+				container := &pod.Spec.InitContainers[i]
+				if quantity, ok := container.Resources.Requests[v1.ResourceEphemeralStorage]; ok {
+					if requestedInit.EphemeralStorage < quantity.Value() {
+						requestedInit.EphemeralStorage = quantity.Value()
+					}
+				}
+			}
+			if requestedInit.EphemeralStorage > requested.EphemeralStorage {
+				requested.EphemeralStorage = requestedInit.EphemeralStorage
+			}
+			return float64((used.EphemeralStorage+requested.EphemeralStorage)*schedulerapi.MaxPriority) / float64(allocatable.EphemeralStorage)
+
+		}
+	default:
+		if v1helper.IsScalarResourceName(resource) {
+			if podRequestsResource(*pod, resource) {
+				for i := range pod.Spec.Containers {
+					container := &pod.Spec.Containers[i]
+					if quantity, ok := container.Resources.Requests[resource]; ok {
+						requested.AddScalar(resource, quantity.Value())
+					}
+				}
+				for i := range pod.Spec.InitContainers {
+					container := &pod.Spec.InitContainers[i]
+					if quantity, ok := container.Resources.Requests[resource]; ok {
+						if requestedInit.ScalarResources[resource] < quantity.Value() {
+							requested.SetScalar(resource, quantity.Value())
+						}
+					}
+
+				}
+				if requestedInit.ScalarResources[resource] > requested.ScalarResources[resource] {
+					requested.ScalarResources[resource] = requestedInit.ScalarResources[resource]
+				}
+
+				return float64((used.ScalarResources[resource]+requested.ScalarResources[resource])*schedulerapi.MaxPriority) / float64(allocatable.ScalarResources[resource])
+
+			}
+		}
+
+	}
+	return 0
+}
+
+// podRequestsResource checks if a given pod requests the scare resource. if false the priority is set to 0
+func podRequestsResource(pod v1.Pod, resource v1.ResourceName) bool {
```

Comment:
just trying to clarify my understanding. Doesnt scarce resource in this function mean any of cpu, mem, ephemeral storage or any extended resource which is requesting this specific priority function ?

Reply:
it would help to rename `resource` as `scarceResource`

Reply:
yes here it can be any resource, the issue with using resource instead of scareResource is an objection from vish on creating a new term to call an existing type. You can look at the comment above from vish.